src/Voron/Data/VoronStream.cs

```diff
@@ -114,10 +131,13 @@ public override int ReadByte()
                     return -1;
             }
 
-            var page = _llt.GetPage(chunk.PageNumber, _pagerRef);
-            _pagerRef.Pager.EnsureMapped(_llt, _pagerRef.PagerPageNumber, VirtualPagerLegacyExtensions.GetNumberOfOverflowPages(page.OverflowSize));
```

Comment:
For sure `EnsureMapped` was called when the page was cached, I'm not sure if we need to repeat it when getting from the cache